### ISSUE TITLE
param-only contexts are improperly considered equal when params differ

### ISSUE THREAD

**Issue by bgentry** [tildeio/router.js#236]
There's a long-running Ember issue (https://github.com/emberjs/ember.js/issues/12419) that discusses a problem where, when using `link-to` with a model's ID instead of the model itself, all links to the same route get the `.ember-transitioning-in` class applied to them, even though their params are different.

I just dug very deep into what was causing this when I encountered it on my app. I wrote up a detailed summary of my findings in this comment, so rather than re-post it here I'll just link to it: https://github.com/emberjs/ember.js/issues/12419#issuecomment-340341755

In short, [`isActiveIntent`](https://github.com/tildeio/router.js/blob/0235c1de0aeb2c6c44215f7ab3ce56e8f4651227/lib/router/router.js#L301) incorrectly returns true in this situation because it is not correctly using the `contexts` that are provided to it when checking route equality.

Though I was able to dig this far, I really have no idea what the appropriate fix is. Hopefully with this information it will be straightforward for somebody who understands router.js.

Cheers :v:

**Comment by rwjblue** [comment]
should have been addressed by #242
